python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -85,7 +82,14 @@ def __init__(self, parent, schema_generation=0, bg_task_manager=None):
 
         super(ShotgunHierarchyModel, self).__init__(parent, bg_task_manager)
 
-        self._hierarchy_is_supported = self._check_hierarchy_is_supported()
+        # check for hierarchy support
+        current_engine = sgtk.platform.current_engine()
+        sg_connection = current_engine.shotgun
+        (self._hierarchy_is_supported, self._hierarchy_not_supported_reason) = \
+            hierarchy_is_supported(sg_connection)
```

Comment:
The demo app was using it, but I think you're right. I've updated based on your suggestion.